### PR TITLE
fix: make broker servicename k8s api compliant[HYB-584]

### PR DESCRIPTION
**Pull request by saumilmac**
Adds:
- Makes broker servicename k8s api character limit compliant

### CODE REVIEW

File: charts/snyk-broker/templates/_helpers.tpl [hunk 1]
```diff
@@ -109,9 +109,12 @@ Create the name of the broker service to use
 */}}
 {{- define "snyk-broker.brokerServiceName" -}}
 {{- if not .Values.disableSuffixes -}}
```
Comment (soniqua) [inline charts/snyk-broker/templates/_helpers.tpl:111]: req: `disableSuffixes` doesn't exist in either `values.yaml` or `values.schema.json` - we should add this and explain what it actually does

File: charts/snyk-broker/templates/_helpers.tpl [hunk 2]
```diff
@@ -109,9 +109,12 @@ Create the name of the broker service to use
...
-{{- .Values.scmType }}-broker-service-{{ .Release.Name }}
+{{- $suffixLength := 16 -}}
```
Comment (soniqua) [inline charts/snyk-broker/templates/_helpers.tpl:112]: req: This is misleading, `suffix` is `.Release.Name` - can this be renamed please? I believe after reading this helper the length of 16 refers to `-broker-service-`.

File: charts/snyk-broker/templates/_helpers.tpl [hunk 3]
```diff
@@ -109,9 +109,12 @@ Create the name of the broker service to use
...
+{{- $releaseNameLength := len .Release.Name -}}
+{{- $truncatedLength := int (sub 63 (add $suffixLength $releaseNameLength)) -}}
+{{- .Values.scmType | trunc $truncatedLength }}-broker-service-{{ .Release.Name }}
```
Comment (soniqua) [inline charts/snyk-broker/templates/_helpers.tpl:115]: req: Please add tests for this

**Review by soniqua — CHANGES_REQUESTED** [review]
Some changes needed here - appreciate this is draft so might still be in progress.

Please make sure we address the CRA service name (as I believe that's the longest service name we might create): `container-registry-agent-broker-service-snyk-broker-container-default`

File: charts/snyk-broker/templates/_helpers.tpl (hunk 1 above)
Comment (saumilmac) [reply charts/snyk-broker/templates/_helpers.tpl:111]: added. I wonder if we still need this variable. do we know if any of our customers uses v1.0.0 of broker ?

File: charts/snyk-broker/templates/_helpers.tpl (hunk 3 above)
Comment (saumilmac) [reply charts/snyk-broker/templates/_helpers.tpl]: added

File: charts/snyk-broker/templates/_helpers.tpl (hunk 2 above)
Comment (saumilmac) [reply charts/snyk-broker/templates/_helpers.tpl]: fixed

**Review by soniqua — APPROVED** [review]
LGTM - I'll follow this up with some suffix cleanup activities

**Comment by soniqua** [comment]
Oh, don't forget to bump the chart version please.

**Pull request merged**